openfe/protocols/restraint_utils/geometry/boresch/geometry.py

```diff
@@ -0,0 +1,298 @@
+      Must be in units compatible with nanometer.
+    host_min_distance : unit.Quantity
+      The minimum distance between any host atom and the guest G0 atom.
+      Must be in units compatible with nanometer.
+    host_max_distance : unit.Quantity
+      The maximum distance between any host atom and the guest G0 atom.
+      Must be in units compatible with nanometer.
+    angle_force_constant : unit.Quantity
+      The force constant for the G1-G0-H0 and G0-H0-H1 angles. Must be
+      in units compatible with kilojoule / mole / radians ** 2.
+    temperature : unit.Quantity
+      The system temperature in units compatible with Kelvin.
+
+    Returns
+    -------
+    BoreschRestraintGeometry
+      An object defining the parameters of the Boresch-like restraint.
+
+    References
+    ----------
+    [1] Baumann, Hannah M., et al. "Broadening the scope of binding free energy
+        calculations using a Separated Topologies approach." (2023).
+    """
+    if (guest_restraint_atoms_idxs is not None) and (host_restraint_atoms_idxs is not None):  # fmt: skip
+        # In this case assume the picked atoms were intentional /
+        # representative of the input and go with it
+        guest_ag = universe.select_atoms[guest_idxs]
+        guest_atoms = [
+            at.ix for at in guest_ag.atoms[guest_restraint_atoms_idxs]
+        ]
+        host_ag = universe.select_atoms[host_idxs]
+        host_atoms = [
+            at.ix for at in host_ag.atoms[host_restraint_atoms_idxs]
+        ]
+
+        # Set the equilibrium values as those of the final frame
+        universe.trajectory[-1]
+        atomgroup = universe.atoms[host_atoms + guest_atoms]
+        bond, ang1, ang2, dih1, dih2, dih3 = _get_restraint_distances(
+            atomgroup
+        )
+
+        # TODO: add checks to warn if this is a badly picked
+        # set of atoms.
+        return BoreschRestraintGeometry(
+            host_atoms=host_atoms,
+            guest_atoms=guest_atoms,
+            r_aA0=bond,
+            theta_A0=ang1,
+            theta_B0=ang2,
+            phi_A0=dih1,
+            phi_B0=dih2,
+            phi_C0=dih3
+        )
+
+    if (guest_restraint_atoms_idxs is not None) ^ (host_restraint_atoms_idxs is not None):  # fmt: skip
+        # This is not an intended outcome, crash out here
+        errmsg = (
+            "both ``guest_restraints_atoms_idxs`` and "
+            "``host_restraint_atoms_idxs`` "
+            "must be set or both must be None. "
+            f"Got {guest_restraint_atoms_idxs} and {host_restraint_atoms_idxs}"
+        )
+        raise ValueError(errmsg)
+
+    # 1. Fetch the guest anchors
+    guest_anchors = find_guest_atom_candidates(
+        universe=universe,
+        rdmol=guest_rdmol,
+        guest_idxs=guest_idxs,
+        rmsf_cutoff=rmsf_cutoff,
+    )
+
+    if len(guest_anchors) == 0:
+        errmsg = "No suitable ligand atoms found for the restraint."
+        raise ValueError(errmsg)
```

Comment:
Do we need an error here since `find_guest_atom_candidates` will raise the same error if no suitable atoms are found in the ligand?

Reply:
Ahh I found an example that can hit this error when we have an atom pool, but all the atoms are collinear in the pool.